Source/Forms/Gen4/Eggs4.cpp

```diff
@@ -326,9 +331,32 @@ void Eggs4::seedToTime()
     time->raise();
 }
 
+void Eggs4::calcPoketch() {
+    QModelIndex index = ui->tableViewGenerator->currentIndex();
+    int advances = generatorModel->data(generatorModel->index(index.row(), 0)).toInt();
+
+    auto *poketch = new Poketch(advances);
+    poketch->show();
+    poketch->raise();
+
+}
+
 void Eggs4::profileManager()
 {
     auto *manager = new ProfileManager4();
     connect(manager, &ProfileManager4::updateProfiles, this, [=] { emit alertProfiles(4); });
     manager->show();
 }
+
+void Eggs4::on_comboBoxGeneratorMethod_currentIndexChanged(int index)
+{
+    auto method = static_cast<Method>(ui->comboBoxGeneratorMethod->currentData().toInt());
+    if(method == Method::Gen4Normal)
+    {
+        calcPoketchGenerator->setVisible(true);
+    }
+    else
+    {
+        calcPoketchGenerator->setVisible(false);
+    }
+}
```

Comment:
You need something similar to this for when the profile changes. I believe poketch is only in DPPt.

Reply:
This also needs to be profile aware. Only update if profile is DPPt.

Reply:
This was added in the second to last commit